src/client/components/Trip/Trip.jsx

```diff
@@ -11,7 +13,45 @@ const Trip = props => {
     members = [];
   }
   members = members.filter((member) => !member.isSelf);
-  const memberNames = members.map((member) => member.name ? member.name.split(' ')[0] : "");
+
+  function renderMemberPhoto(user) {
+    return (
+      <div className="member-photo">
+       { user.profile_photo
+         ? (<img src={user.profile_photo}></img>)
+         : <ProfileButton />
+       }
+      </div>
+    );
+  }
+
+  function renderMemberPhotos() {
+    if (members.length <= 4) {
+      return members.map(renderMemberPhoto);
+    } else {
+      return members.slice(0, 3).map(renderMemberPhoto).concat([
+        <div className="member-photo">
+          <IconButton className="extra-members-photo" icon={"+" + (members.length - 3)} />
+        </div>
+      ]);
+    }
+  }
+
+  function renderMemberFirstName(user) {
+    return user.name
+      ? user.name.split(' ')[0]
```

Comment:
For guest users, the name will just be _GUEST_USER. Can we add a case to handle that? Maybe we can show something like "Anon" or "Guest" or "Guest 1", etc. 

In the group page, I just displayed it as "Anon Traveller 1 " or something like that